### PR TITLE
Fix SFL race condition

### PR DESCRIPTION
**Pull request by OliverJAsh**
Sometimes the header count would not appear, because this code ran before the template insertion due to our nasty event system.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/save-for-later.js [hunk 1]
```diff
@@ -73,6 +75,16 @@ define([
 
     SaveForLater.prototype.init = function () {
         var userLoggedIn = identity.isUserLoggedIn();
+
+        var popup = qwery('.popup--profile')[0];
+        fastdom.write(function () {
+            bonzo(popup).prepend(bonzo.create(
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/save-for-later.js]: `$`

File: static/src/javascripts/projects/common/modules/experiments/tests/save-for-later.js [hunk 2]
```diff
@@ -44,20 +42,21 @@ define([
...
-                                    idUrl: config.page.idUrl
-                                })
-                            ));
+                    var loadProfileNav = function () {
+                        return new Promise(function (resolve, reject) {
+                            mediator.on('modules:profilenav:loaded', resolve);
                         });
+                    };
+
+                    Promise.all([loadIdentityApi(), loadProfileNav()]).then(function () {
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/experiments/tests/save-for-later.js]: why are these functions?

File: static/src/javascripts/projects/common/modules/experiments/tests/save-for-later.js [hunk 3]
```diff
@@ -44,20 +42,21 @@ define([
             {
                 id: 'variant',
                 test: function () {
-                    mediator.on('module:identity:api:loaded', function () {
-                        var saveForLater = new SaveForLater();
-                        saveForLater.init();
-                    });
+                    var loadIdentityApi = function () {
+                        return new Promise(function (resolve, reject) {
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/experiments/tests/save-for-later.js]: unused `reject`

File: static/src/javascripts/projects/common/modules/experiments/tests/save-for-later.js [hunk 4]
```diff
@@ -44,20 +42,21 @@ define([
...
-                    mediator.on('modules:profilenav:loaded', function () {
-                        var popup = qwery('.popup--profile')[0];
-                        fastdom.write(function () {
-                            bonzo(popup).prepend(bonzo.create(
-                                template(profileLinkTmp.replace(/^\s+|\s+$/gm, ''), {
-                                    idUrl: config.page.idUrl
-                                })
-                            ));
+                    var loadProfileNav = function () {
+                        return new Promise(function (resolve, reject) {
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/experiments/tests/save-for-later.js]: unused `reject`

File: static/src/javascripts/projects/common/modules/save-for-later.js (hunk 1 above)
Comment (OliverJAsh) [inline static/src/javascripts/projects/common/modules/save-for-later.js]: I wonder if we could create a linter rule for this?

Comment (OliverJAsh) [inline static/src/javascripts/projects/common/modules/save-for-later.js]: P.S. This is not a bonzo element, so no `$`.

Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/save-for-later.js]: `$=bonzo(qwery())` so this would have been valid

**Pull request merged**